bubblewrap.c

```diff
@@ -540,6 +540,35 @@ get_oldroot_path (const char *path)
   return strconcat ("/oldroot/", path);
 }
 
+static char *
+generate_map_file (bool map_root, uid_t sandbox_id, uid_t parent_id)
+{
+  const int INITIAL_UNKNOWN_ID = 10000;
+  cleanup_free char *line1 = NULL;
+  cleanup_free char *line2 = NULL;
+  cleanup_free char *line3 = NULL;
+  cleanup_free char *line4 = NULL;
+  if (map_root && parent_id != 0 && sandbox_id != 0)
+    {
+        line1 = xasprintf ("0 0 1\n");
+    }
+  if (opt_unshare_user)
+    {
+      if (sandbox_id > 999)
+          line2 = xasprintf ("%d %d %d\n", map_root ? 1 : 0, INITIAL_UNKNOWN_ID, 1000 - (map_root ? 1 : 0));
+      else
+        {
+          int size_first_interval = sandbox_id - (map_root ? 1 : 0);
+          if (size_first_interval > 1)
+            line2 = xasprintf ("%d %d %d\n", map_root ? 1 : 0, INITIAL_UNKNOWN_ID, size_first_interval);
+          if (1000 - size_first_interval > 2)
+            line4 = xasprintf ("%d %d %d\n", sandbox_id + 1, INITIAL_UNKNOWN_ID + size_first_interval, 1000 - size_first_interval - 2);
+        }
+    }
+  line3 = xasprintf ("%d %d 1\n", sandbox_id, parent_id);
+  return strconcat4 (line1, line2, line3, line4);
+}
```

Comment:
This breaks any use of non-privileged user namespaces use, because that only allows a single mapping.